samples/sample_encode/src/sample_encode.cpp

```diff
@@ -1405,6 +1409,14 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
         }
     }
 
+    if(!pParams->memType) {
+#if defined(_WIN32) || defined(_WIN64)
+        pParams->memType = pParams->bUseHWLib ? D3D11_MEMORY : SYSTEM_MEMORY;
+#else
+        pParams->memType = SYSTEM_MEMORY;
```

Comment:
Why different behavior on Linux and Windows? Should be the same.

Reply:
Please pay attention to system memory case? E.g. https://github.com/Intel-Media-SDK/MediaSDK/blob/0be32d7f99f7e1eef6c4329f4892e5e0f14128b1/samples/sample_encode/src/pipeline_encode.cpp#L911 
CD3D9Device will be used but we need D3D11.

Reply:
Similar place here: https://github.com/Intel-Media-SDK/MediaSDK/blob/0be32d7f99f7e1eef6c4329f4892e5e0f14128b1/samples/sample_encode/src/pipeline_encode.cpp#L1632